### ISSUE TITLE
MathML

### ISSUE THREAD

**Issue by hlinander** [dmjio/miso#325]
Tried to output mathml using `nodeHtml` to produce the child elements, however even if the dom looked correct on inspection with dev tools the browser does not render the content correctly. I suppose this is due to the VDom handling? I noticed the `_math` element existed so perhaps I am missing something.

**Comment by dmjio** [comment]
@hlinander thanks for bringing this up, this will make a cool example. Firstly, can you paste the code in question that isn't being rendered properly?

**Comment by hlinander** [comment]
For example, using the code from sample-app but replacing the viewModel with
```
viewModel _ = math_ [] [
  nodeHtml "msup" [] [
      nodeHtml "mi" [] [ text "x" ],
      nodeHtml "mn" [] [ text "2" ]
                     ]
    ]
```
generates a DOM of the form

`<math><msup><mi>x</mi><mn>2</mn></msup></math>`

which is correct but which does not render as it should (https://jsfiddle.net/vdumtzsx/), i.e. the exponent is not placed correctly but rather just following the base.

I have tried this in Firefox 57.0.1 and Chromium 63.

**Comment by hlinander** [comment]
Probably the same issue as https://github.com/reflex-frp/reflex/issues/38,
i.e. these elements need to be created in the correct namespace. This happens automatically when loading a HTML page but not when injected via `document.createElement`, instead `document.createElementNS` is needed. Is it possible to specify the XML namespace for an element with miso?

**Comment by dmjio** [comment]
All DOM elements are defined in terms of `node`
```haskell
node :: NS -> MisoString -> Maybe Key -> [Attribute m] -> [View m] -> View m
```

Right now I believe only SVG and HTML namespaces are supported. 
https://github.com/dmjio/miso/blob/master/jsbits/diff.js#L136-L138

I think it might be possible just specify 
```xml
<math xmlns="http://www.w3.org/1998/Math/MathML">
```
in `miso` as
```haskell
math_ = node HTML "math" [ prop "xmlns" "http://www.w3.org/1998/Math/MathML" ] 
```

Also, the fact Chrome does not support it, we'd have to ask if it's worth it. There are other solutions like `MathJAX` which I think provides fallbacks for Chrome. I am not as familiar with MathJAX though, but it should be possible to integrate.

**Comment by dmjio** [comment]
@hlinander any updates on this?

**Comment by dmjio** [comment]
closing due to inactivity

**Comment by hlinander** [comment]
@dmjio: Hope it is ok to resurrect this issue :)

The prop "xmlns" approach does not work since the crux is the JS createElement call that needs to be a createElementNS with the correct namespace.
See PR #439 where I tried to follow the SVG implementation.

Regarding Chrome support I agree that MathJax is a good fallback (and can be added very simply by just loading it from a <script> tag in index, but this should be up to the user and not something Miso should take care of right?). I have used it extensively for LaTeX rendering but not for MML although I believe that should work out of the box.
